File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/adapters/BaseAdapter.java
package ru.testsimpleapps.coloraudioplayer.ui.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.View;


public abstract class BaseAdapter extends RecyclerView.Adapter {

    protected static final int TYPE_HEADER = 0;
    protected static final int TYPE_ITEM = 1;

    public interface OnItemClickListener {
        void onItemClick(View view, int position);
    }

    public interface OnItemLongClickListener {
        void onItemLongClick(View view, int position);
    }

    public interface OnItemCheckListener {
        void onItemCheck(View view, int position);
    }

    protected OnItemClickListener mOnItemClickListener;
    protected OnItemLongClickListener mOnItemLongClickListener;
    protected OnItemCheckListener mOnItemCheckListener;

    public void setOnItemClickListener(OnItemClickListener listener) {
        mOnItemClickListener = listener;
    }

    public void setOnItemLongClickListener(OnItemLongClickListener listener) {
        mOnItemLongClickListener = listener;
    }

    public void setOnItemCheckListener(OnItemCheckListener listener) {
        mOnItemCheckListener = listener;
    }

}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/adapters/BaseAdapter.kt
package ru.testsimpleapps.coloraudioplayer.ui.adapters

import android.support.v7.widget.RecyclerView
import android.view.View


abstract class BaseAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    protected lateinit var mOnItemClickListener: OnItemClickListener
    protected lateinit var mOnItemLongClickListener: OnItemLongClickListener
    protected lateinit var mOnItemCheckListener: OnItemCheckListener

    interface OnItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    interface OnItemLongClickListener {
        fun onItemLongClick(view: View, position: Int)
    }

    interface OnItemCheckListener {
        fun onItemCheck(view: View, position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        mOnItemClickListener = listener
    }

    fun setOnItemLongClickListener(listener: OnItemLongClickListener) {
        mOnItemLongClickListener = listener
    }

    fun setOnItemCheckListener(listener: OnItemCheckListener) {
        mOnItemCheckListener = listener
    }

    companion object {

        val TYPE_HEADER = 0
        val TYPE_ITEM = 1
    }

}